server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java

```diff
@@ -246,10 +247,20 @@ public String loginForHtml(Model model, Principal principal, HttpServletRequest
         return Arrays.asList(ofNullable(cookies).orElse(new Cookie[]{}))
                 .stream()
                 .filter(c -> c.getName().startsWith("Saved-Account"))
-                .map(c -> JsonUtils.readValue(c.getValue(), clazz))
+                .map(c -> JsonUtils.readValue(decodeCookieValue(c.getValue()), clazz))
                 .collect(Collectors.toList());
     }
 
+    private static String decodeCookieValue(String inValue) {
+        String out = null;
+        try {
+            out = URLDecoder.decode(inValue, UTF_8.name());
+        } catch (Exception e) {
+            return "";
```

Comment:
does `""` deserialize as `clazz`, or will an error here, return `""` cause the entire list to fail in the calling method?

and if that is the case, no logging on an exception here, not even debug, would cause us to lose the root cause of the issue, and become very hard to track down.